pipelines/broad/internal/arrays/imputation/BroadInternalImputation.wdl

```diff
@@ -0,0 +1,238 @@
+            tdr_target_table_name   = "ImputationWideOutputsTable",
+            outputs_tsv             = format_imputation_wide_outputs.ingest_outputs_wide_tsv
+    }
+
+    output {
+        File aggregated_imputation_metrics              = Imputation.aggregated_imputation_metrics
+        File chunks_info                                = Imputation.chunks_info
+        File failed_chunks                              = Imputation.failed_chunks
+        File imputed_multisample_vcf                    = Imputation.imputed_multisample_vcf
+        File imputed_multisample_vcf_index              = Imputation.imputed_multisample_vcf_index
+        Array[File]? imputed_single_sample_vcfs         = Imputation.imputed_single_sample_vcfs
+        Array[File]? imputed_single_sample_vcf_indices  = Imputation.imputed_single_sample_vcf_indices
+        File n_failed_chunks                            = Imputation.n_failed_chunks
+    }
+}
+
+task format_imputation_outputs {
+    input {
+        String          aggregated_imputation_metrics
+        String          chunks_info
+        String          failed_chunks
+        String          imputed_multisample_vcf
+        String          imputed_multisample_vcf_index
+        Array[String]?  imputed_single_sample_vcfs
+        Array[String]?  imputed_single_sample_vcf_indices
+        Array[String]   chip_well_barcodes
+        String          n_failed_chunks
+    }
+
+    command <<<
+
+        # write header to file
+        echo -e "aggregated_imputation_metrics\tchunks_info\tfailed_chunks\tn_failed_chunks\t\
+        imputed_multisample_vcf\timputed_multisample_vcf_index\t\
+        imputed_single_sample_vcfs\timputed_single_sample_vcf_indices\t\
+        chip_well_barcodes" \
+        > ingestDataset_imputation_outputs.tsv
+
+        # handle array[type] variables to print as list with double quotes
+        imputed_single_sample_vcfs='~{sep='","' imputed_single_sample_vcfs}'
+        imputed_single_sample_vcf_indices='~{sep='","' imputed_single_sample_vcf_indices}'
+        chip_well_barcodes='~{sep='","' chip_well_barcodes}'
+
+        # write file paths to row in tsv file
+        echo -e "~{aggregated_imputation_metrics}\t~{chunks_info}\t~{failed_chunks}\t~{n_failed_chunks}\t\
+        ~{imputed_multisample_vcf}\t~{imputed_multisample_vcf_index}\t\
+        [\"${imputed_single_sample_vcfs}\"]\t\
+        [\"${imputed_single_sample_vcf_indices}\"]\t\
+        [\"${chip_well_barcodes}\"]" \
+        >> ingestDataset_imputation_outputs.tsv
+
+
+        python3 << CODE
+        import pandas as pd
+
+        tsv_df = pd.read_csv("ingestDataset_imputation_outputs.tsv", sep="\t")
+        tsv_df = tsv_df.dropna(axis=1, how="all")  # drop columns if no value (optional outputs etc)
+
+        outputs = tsv_df.to_json("ingestDataset_imputation_outputs.json", orient="records")  # write json file
+
+        CODE
+    >>>
+
+    runtime {
+        docker: "broadinstitute/horsefish"
```

Comment:
we usually want to have a tag for our docker images that is semantic or versionable, this will always just pull the latest tag which could have unexpected results